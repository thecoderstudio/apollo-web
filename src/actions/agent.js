export const LIST_AGENTS = 'LIST_AGENTS';

export const listAgents = (agents) => ({
  type: LIST_AGENTS,
  agents
});