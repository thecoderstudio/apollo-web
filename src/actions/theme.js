function changeTheme(theme) {
  return {
    type: 'CHANGE_THEME',
    theme
  };
};

export { changeTheme };
