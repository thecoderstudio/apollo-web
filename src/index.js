import React from 'react';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'
import { render } from 'react-dom';
import App from './App';
import { store, persistor } from './store';

render(
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <App />
    </PersistGate>
  </Provider>,
  document.getElementById('app')
);
