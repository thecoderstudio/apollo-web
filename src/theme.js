export const darkTheme = {
  name: "dark",
  black: "#181818",
  lightBlack: "#1B1B1B",
  white: "#F8F8F8",
  primary: "#02A3EE",
  accent: "#4D02EE",
  connectedColor: "#27AE60",
  disconnectedColor: "#C0392B",
  connectingColor: "#F39C12",
  error: "#ED3E2A"
};

function getBrowserPreferredTheme() {
  return darkTheme;
}

export const browserPreferredTheme = getBrowserPreferredTheme();
