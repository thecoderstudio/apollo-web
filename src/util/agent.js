export function getFontAwesomeClass(operatingSystem) {
  switch (operatingSystem) {
      case 'darwin':
        return "fab fa-apple";
      case 'linux':
        return "fab fa-linux";
      case 'freebsd':
        return "fab fa-freebsd";
      default:
        return "fas fa-question-circle";
    }
}
