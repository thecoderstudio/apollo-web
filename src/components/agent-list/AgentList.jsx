import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import AgentListItem from './AgentListItem';
import { listAgents as listAgentsAction } from '../../actions/agent';

const Content = styled.div`
  display: grid;
  grid-template-rows: [title] 50px [list] 1fr;

  background-color: ${props => props.theme.lightBlack};
  border-radius: 8px;
  maring: 20px;

  padding: 20px;
`;

const ListTitle = styled.h2`
  grid-row: title;
`;

const List = styled.ul`
  grid-row: list;
  list-style: none;
  padding-left: 0;
`;

class AgentList extends React.Component {
  constructor(props) {
    super(props);
    this.generateAgents = this.generateAgents.bind(this);
    this.dispatchListAgents = this.dispatchListAgents.bind(this);
    this.setupWebSocket();
  }

  setupWebSocket() {
    let server = new WebSocket(`${process.env.APOLLO_WS_URL}agent`);
    server.onmessage = (event) => {
      this.dispatchListAgents(event.data);
    };
  }

  generateAgents(agents) {
    return agents.map(agent => {
      return <AgentListItem key={agent.id} agentName={agent.name} connectionState={agent.connection_state} />;
    });
  }

  dispatchListAgents(data) {
    const { dispatch } = this.props;
    dispatch(listAgentsAction(JSON.parse(data)));
  }

  render() {
    return (
      <Content className={this.props.className}>
        <ListTitle>Agents</ListTitle>
        <List>
          {this.generateAgents(this.props.agents)}
        </List>
      </Content>
    );
  }
}

export default connect(
  state => ({agents: state.agent})
)(AgentList);