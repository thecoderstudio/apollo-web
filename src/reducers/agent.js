import { LIST_AGENTS } from '../actions/agent';

const initialState = {
    agents: [],
};

export default function agentReducer(state = initialState, action) {
    switch (action.type) {
        case LIST_AGENTS:
            console.log(
                {
                    ...state,
                    agents: action.agents,
                }
            )
            return {
                ...state,
                agents: action.agents,
            };
        default: return state;
    }
}