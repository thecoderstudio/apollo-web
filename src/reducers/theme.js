import { browserPreferredTheme } from '../theme';

export default function themeReducer(state = { theme: browserPreferredTheme }, action) {
  switch (action.type) {
    case 'CHANGE_THEME':
      return { theme: action.theme };
    default:
      return state;
  }
}
