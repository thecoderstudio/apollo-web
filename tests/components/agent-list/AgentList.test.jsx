import React from 'react';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import WS from 'jest-websocket-mock';
import { ThemeProvider } from 'styled-components';
import AgentList from '../../../src/components/agent-list/AgentList';
import { listAgents } from '../../../src/actions/agent';
import waitForExpect from 'wait-for-expect';
import { darkTheme } from '../../../src/theme';


const mockStore = configureStore([]);

function getComponent(store) {
  return renderer.create(
    <Provider store={store}>
      <ThemeProvider theme={darkTheme}>
        <AgentList />
      </ThemeProvider>
    </Provider>
  );
}

describe('agentList', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      authenticated: true,
      agent: []
    });
    process.env = {
      APOLLO_WS_URL: 'ws://localhost:1234/'
    };
    store.dispatch = jest.fn();
  });

  afterEach(() => {
    WS.clean();
  });

  it("renders correctly", () => {
    let tree = getComponent(store).toJSON();
    expect(tree).toMatchSnapshot()
  });

  it("correctly dispatches list agents", async () => {
    const data = [
      { id: "id", name: "name", connection_state: "connected" },
      { id: "id2", name: "name", connection_state: "connected" },
    ];

    const server = new WS(`ws://localhost:1234/agent`, { jsonProtocol: true });
    getComponent(store);

    await server.connected;

    server.send(data);

    await waitForExpect(() => {
      expect(store.dispatch).toHaveBeenCalledWith(
        listAgents(data)
      );
    });
  });


  it("correctly lists multiple agents", () => {
    store = mockStore({
      authenticated: true,
      agent: [
        { id: "id", name: "name", connection_state: "connected" },
        { id: "id2", name: "name", connection_state: "connected" },
      ]
    })
    expect(getComponent(store)).toMatchSnapshot();
  })
})
