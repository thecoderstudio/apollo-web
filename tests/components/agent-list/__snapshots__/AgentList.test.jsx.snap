// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`agentList correctly lists multiple agents 1`] = `
.c8 {
  display: grid;
  grid-template-columns: [status-indicator] 20px [status-text] 1fr;
}

.c9 {
  grid-column: status-indicator;
  height: 12px;
  width: 12px;
  margin: 10px;
  border-radius: 50%;
  float: right;
  text-align: left;
  color: #27AE60;
  background-color: #27AE60;
}

.c10 {
  grid-column: status-text;
  position: relative;
  margin: 0;
  padding-top: 0;
  margin-left: 15px;
  text-align: left;
  color: #27AE60;
}

.c3 {
  display: grid;
  grid-template-columns: 1fr 1fr 175px;
  grid-column-gap: 10px;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 8px;
  border: 1px solid white;
  height: 30px;
  line-height: 30px;
  padding: 15px 20px;
  margin-top: 25px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  width: 50%;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c6 {
  width: 50%;
}

.c11 {
  cursor: pointer;
  color: #F8F8F8;
}

.c5 {
  cursor: pointer;
  margin-right: 10px;
  color: #F8F8F8;
  cursor: inherit;
}

.c0 {
  display: grid;
  grid-template-rows: [title] 50px [list] 1fr;
  background-color: #1B1B1B;
  border-radius: 8px;
  padding: 20px;
}

.c1 {
  grid-row: title;
}

.c2 {
  grid-row: list;
  list-style: none;
  padding-left: 0;
}

@media (max-width:47.9375em) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    height: auto;
  }
}

@media (max-width:47.9375em) {
  .c7 {
    width: 100%;
  }
}

@media (max-width:47.9375em) {
  .c4 {
    width: 100%;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    text-align: center;
  }
}

@media (max-width:47.9375em) {
  .c6 {
    width: 100%;
    text-align: center;
  }
}

<div
  className="c0"
>
  <h2
    className="c1"
  >
    Agents
  </h2>
  <ul
    className="c2"
  >
    <li
      className="c3"
    >
      <div
        className="c4"
      >
        <i
          className="c5 fas fa-question-circle"
          data-tip="Operating system unkown"
        />
        <p>
          name
        </p>
      </div>
      <p
        className="c6"
        data-tip="External IP Address"
      />
      <div
        className="c7"
      >
        <div
          className="c8"
        >
          <div
            className="c9"
          />
          <p
            className="c10"
          >
            Connected
          </p>
        </div>
        <i
          className="c11 fas fa-terminal"
          onClick={[Function]}
        />
      </div>
      <div
        className="__react_component_tooltip t30cd0a5c-fa6a-40c7-bc16-ab4b2bf9cd6e place-top type-dark"
        data-id="tooltip"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": "
  	.t30cd0a5c-fa6a-40c7-bc16-ab4b2bf9cd6e {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.t30cd0a5c-fa6a-40c7-bc16-ab4b2bf9cd6e.place-top {
        margin-top: -10px;
    }
    .t30cd0a5c-fa6a-40c7-bc16-ab4b2bf9cd6e.place-top::before {
        border-top: 8px solid transparent;
    }
    .t30cd0a5c-fa6a-40c7-bc16-ab4b2bf9cd6e.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .t30cd0a5c-fa6a-40c7-bc16-ab4b2bf9cd6e.place-bottom {
        margin-top: 10px;
    }
    .t30cd0a5c-fa6a-40c7-bc16-ab4b2bf9cd6e.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .t30cd0a5c-fa6a-40c7-bc16-ab4b2bf9cd6e.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .t30cd0a5c-fa6a-40c7-bc16-ab4b2bf9cd6e.place-left {
        margin-left: -10px;
    }
    .t30cd0a5c-fa6a-40c7-bc16-ab4b2bf9cd6e.place-left::before {
        border-left: 8px solid transparent;
    }
    .t30cd0a5c-fa6a-40c7-bc16-ab4b2bf9cd6e.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .t30cd0a5c-fa6a-40c7-bc16-ab4b2bf9cd6e.place-right {
        margin-left: 10px;
    }
    .t30cd0a5c-fa6a-40c7-bc16-ab4b2bf9cd6e.place-right::before {
        border-right: 8px solid transparent;
    }
    .t30cd0a5c-fa6a-40c7-bc16-ab4b2bf9cd6e.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  ",
            }
          }
        />
      </div>
    </li>
    <li
      className="c3"
    >
      <div
        className="c4"
      >
        <i
          className="c5 fas fa-question-circle"
          data-tip="Operating system unkown"
        />
        <p>
          name2
        </p>
      </div>
      <p
        className="c6"
        data-tip="External IP Address"
      />
      <div
        className="c7"
      >
        <div
          className="c8"
        >
          <div
            className="c9"
          />
          <p
            className="c10"
          >
            Connected
          </p>
        </div>
        <i
          className="c11 fas fa-terminal"
          onClick={[Function]}
        />
      </div>
      <div
        className="__react_component_tooltip t0d4a81c9-30a9-4e1b-9d06-44580c38baa9 place-top type-dark"
        data-id="tooltip"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": "
  	.t0d4a81c9-30a9-4e1b-9d06-44580c38baa9 {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.t0d4a81c9-30a9-4e1b-9d06-44580c38baa9.place-top {
        margin-top: -10px;
    }
    .t0d4a81c9-30a9-4e1b-9d06-44580c38baa9.place-top::before {
        border-top: 8px solid transparent;
    }
    .t0d4a81c9-30a9-4e1b-9d06-44580c38baa9.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .t0d4a81c9-30a9-4e1b-9d06-44580c38baa9.place-bottom {
        margin-top: 10px;
    }
    .t0d4a81c9-30a9-4e1b-9d06-44580c38baa9.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .t0d4a81c9-30a9-4e1b-9d06-44580c38baa9.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .t0d4a81c9-30a9-4e1b-9d06-44580c38baa9.place-left {
        margin-left: -10px;
    }
    .t0d4a81c9-30a9-4e1b-9d06-44580c38baa9.place-left::before {
        border-left: 8px solid transparent;
    }
    .t0d4a81c9-30a9-4e1b-9d06-44580c38baa9.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .t0d4a81c9-30a9-4e1b-9d06-44580c38baa9.place-right {
        margin-left: 10px;
    }
    .t0d4a81c9-30a9-4e1b-9d06-44580c38baa9.place-right::before {
        border-right: 8px solid transparent;
    }
    .t0d4a81c9-30a9-4e1b-9d06-44580c38baa9.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  ",
            }
          }
        />
      </div>
    </li>
  </ul>
</div>
`;

exports[`agentList renders correctly 1`] = `
.c0 {
  display: grid;
  grid-template-rows: [title] 50px [list] 1fr;
  background-color: #1B1B1B;
  border-radius: 8px;
  padding: 20px;
}

.c1 {
  grid-row: title;
}

.c2 {
  grid-row: list;
  list-style: none;
  padding-left: 0;
}

<div
  className="c0"
>
  <h2
    className="c1"
  >
    Agents
  </h2>
  <ul
    className="c2"
  />
</div>
`;
